package-res/resources/web/pentaho/type/Context.js

```diff
@@ -155,15 +240,63 @@ define([
     },
 
     /**
-     * Gets the instance constructor of a type.
+     * Gets, asynchronously, the **configured instance constructor** instance constructor of a type.
+     *
+     * For more information on the `typeRef` argument,
+     * please see [UTypeReference]{@link pentaho.type.spec.UTypeReference}.
+     *
+     * This method can be used even if a generic type metadata specification references
+     * non-standard types whose modules have not yet been loaded by the AMD module system.
+     *
+     * An error is thrown when:
```

Comment:
Let me rewrite the aforementioned rules:
1. `argInvalidType` and `argRequired`
  * already documented in an argument's tag
  * sometimes these are actually implemented and thrown — but simply because it helps catch common programming bugs faster
  * a `@throws` clause would be redundant and would document a circumstantial extra-precaution being taken
  * if we implement the error, we need to test it as well (or coverage drops), but that doesn't mean we need to document it
  * when implemented, these are like **assertions**
  * if the code were compiled, most cases would be caught at compile time
  * if the code were compiled, most errors would never reach a try/catch
  * there is a gotcha in these two statements: to catch _nully_ at compile time, the compiler would need to perform "null analysis", something which is hard to perform
  * however, the main and relevant distinction is between good use and bad use of a contract — errors thrown in the course of good/acceptable/unforseeable use are documented and can/should be handled by callers, while errors thrown in the course of bad use (invalid program) are not.
  * this is akin to how in Java only some types of errors are required to be part of method signatures
  * last, but not the least, it would be overkill to document these errors; it would be lengthy, and it would actually feel redundant.
2. `argInvalid` and `operInvalid`
  * document using `@throws`
  * implement, using `throw`
  * these conditions depend on the flow of the program and the state of its objects and are utterly impossible to detect at compile time
  * mostly these will still be considered the result of a "bad program"
3. "timeout", "user validation", ...?
  * document using `@throws`
  * implement, using `throw`
  * these conditions generally depend on state and actions that are external to the program and whose occurrence it thus can't control or foresee, but only expect and prepare for

Questions that I still think need discussion is:
* how to handle the difference between rejections and throws; there's an extensive discussion on this subject in https://github.com/jsdoc3/jsdoc/issues/509
* should we start using `@throws` now, or do it all in a full replace do the codebase?